extra/2-piping.js

```diff
@@ -16,26 +16,43 @@
   the final result to the variable goodCode
 */
 
-function add() {
-
+function add(num1, num2) {
+  return num1+num2;
 }
 
-function multiply() {
-
+function multiply(num1, num2) {
+  return num1*num2;
 }
 
-function format() {
-
+function format(value) {
+  return "£"+value;
 }
 
 const startingValue = 2;
 
 // Why can this code be seen as bad practice? Comment your answer.
-let badCode =
+//declare let after the function it's a bad practice
+
+function addAndFormated1(value) {
+  soma = value + 10;
+  total = soma * 2;
+  formatado = "£" +total;
+  return formatado;
+}
+
+let badCode = addAndFormated1(startingValue);
 
 /* BETTER PRACTICE */
+//declare let before the function it's a good practice
+let goodCode = addAndFormated2(startingValue);
 
-let goodCode =
+
+function addAndFormated2(value) {
+  var soma = value + 10;
+  var total = soma * 2;
+  var formatado = "£" +total;
+  return formatado;
+}
```

Comment:
You should make use of the 3 functions add, multiply and format, and call them all in sequence rather than putting all the functionality in one method. This makes the code easier to understand and makes the methods more reusable.

Reply:
Done